File: blind75/php/besttimetobuyandsellstock.php
class Solution {

    /**
     * @param Integer[] $prices
     * @return Integer
     */
    function maxProfit($prices) {
        if (empty($prices)) {
            return 0;
        }
        
        $maxProfit = 0;
        $minPrice = $prices[0];

        foreach ($prices as $price) {
            $maxProfit = max($maxProfit, $price - $minPrice);
            $minPrice = min($minPrice, $price);
        }

        return $maxProfit;
    }
}

File: blind75/c++/besttimetobuyandsellstock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.empty()) {
            return 0;
        }

        int maxProfit = 0;
        int minPrice = prices[0];

        for (int i = 1; i < prices.size(); i++) {
            maxProfit = max(maxProfit, prices[i] - minPrice);
            minPrice = min(minPrice, prices[i]);
        }

        return maxProfit;
    }
};
